src/init.cpp

```diff
@@ -1277,6 +1283,12 @@ bool AppInit2(Config &config, thread_group &threadGroup)
 #endif // !ENABLE_WALLET
 
     // ********************************************************* Step 8: data directory maintenance
+    if (GetBoolArg("-txindex", DEFAULT_TXINDEX))
+    {
+        auto txindex_db = std::make_unique<TxIndexDB>(nTxIndexCache, false, fReindex);
+        g_txindex = std::make_unique<TxIndex>(std::move(txindex_db));
```

Comment:
fixed.

Reply:
I didn't write this code but actually I think this is correct.  Both the g_txindex and the txindexdb are unique pointers...the txindexdb being the db inside the g_txindex object.  There is no way around it unless we entirely abandon the smart pointer for the txindexdb and then we'll just be back to using a regular pointer with a delete on destruction without the benefit of the smart pointer.  I would prefer to keep the smart pointer because if we in the future have other smart pointers (which seems to be the trend within the codebase) within the db object then they'll get destructed in the right order..i've painfully been through this with graphene and is why I switched all the sub-objects there to smart pointers.

Reply:
I still have 2 smart pointers (TxIndex member and g_txindex) .  The original code had 4.  The 2 you are not seeing is the temporary created in the if GetBoolArg scope and the temporary parameter of the TxIndex ctor.

Reply:
ok you're right,  done.